src/SmartComponents/Reports/WorkloadsDetectedTable/WorkloadsDetectedTable.tsx

```diff
@@ -0,0 +1,335 @@
+    EmptyStateIcon,
+    EmptyStateVariant,
+    Title,
+    TitleLevel,
+    EmptyStateBody,
+    Stack,
+    StackItem,
+    Card,
+    CardBody
+} from '@patternfly/react-core';
+import { ErrorCircleOIcon, SearchIcon } from '@patternfly/react-icons';
+import { WorkloadModel } from '../../../models';
+import { ObjectFetchStatus } from '../../../models/state';
+import debounce from 'lodash/debounce';
+import { formatNumber } from '../../../Utilities/formatValue';
+import './WorkloadsDetectedTable.scss';
+import { isNullOrUndefined } from '../../../Utilities/formUtils';
+
+interface StateToProps extends RouterGlobalProps {
+    reportWorkloadsDetected: {
+        total: number;
+        items: WorkloadModel[]
+    };
+    reportWorkloadsDetectedFetchStatus: ObjectFetchStatus;
+}
+
+interface DispatchToProps {
+    fetchReportWorkloadsDetected: (
+        reportId: number,
+        page: number,
+        perPage: number,
+        orderBy: string | undefined,
+        orderDirection: 'asc' | 'desc' | undefined
+    ) => any;
+}
+
+interface Props extends StateToProps, DispatchToProps {
+    reportId: number;
+};
+
+interface Column extends ICell {
+    key: string;
+    cellFormatters?: any;
+    transforms?: any;
+}
+
+interface State {
+    page: number;
+    perPage: number;
+    columns: Column[];
+    rows: IRow[];
+    sortBy: ISortBy;
+};
+
+class WorkloadsDetectedTable extends React.Component<Props, State> {
+
+    public changePage = debounce(() => {
+        this.refreshData();
+    }, 800);
+
+    constructor(props: Props) {
+        super(props);
+        this.state = {
+            page: 1,
+            perPage: 10,
+            columns: [
+                {
+                    title: 'Workloads',
+                    key: 'workload',
+                    props: { },
+                    transforms: [ cellWidth('25'), sortable ]
+                },
+                {
+                    title: 'OS',
+                    key: 'osName',
+                    props: { },
+                    transforms: [ cellWidth('25'), sortable ]
+                },
+                {
+                    title: 'In Clusters',
+                    key: 'clusters',
+                    props: { },
+                    transforms: [ cellWidth('25') ]
+                },
+                {
+                    title: 'VMs',
+                    key: 'vms',
+                    props: { },
+                    transforms: [ cellWidth('25'), sortable ]
+                }
+            ],
+            rows: [],
+            sortBy: { }
+        };
+    }
+
+    public componentDidMount() {
+        this.refreshData();
+    }
+
+    public refreshData = (
+        page: number = this.state.page,
+        perPage: number = this.state.perPage,
+        { direction, index } = this.state.sortBy
+    ) => {
+        const { reportId, fetchReportWorkloadsDetected } = this.props;
+
+        const column = index ? this.state.columns[index - 1].key : undefined;
```

Comment:
i feel like the `- 1` is wrong because testing the UI the sorting is making the remote call to the backend with the wrong `orderBy` value.
```suggestion
        const column = index ? this.state.columns[index].key : undefined;
```

Reply:
@mrizzi thanks for discovering this problem. `[index - 1]` should only be used when we are in an expandable table because it adds an extra column. Since this is a normal table we should use `[index]`. Thanks

Reply:
thanks for the explanation @carlosthe19916